File: sim_mqtt.js
const mqtt = require("mqtt");

const broker = "wss://broker.emqx.io:8084/mqtt";
const topic = "mi_proyecto/sensor/rpm";

const options = {
    reconnectPeriod: 2000,
    connectTimeout: 4000,
};

const client = mqtt.connect(broker, options);

client.on("connect", () => {
    console.log("Connected to MQTT broker for simulation.");

    let rpm = 120.0;

    setInterval(() => {
        // Generate a new RPM value that changes slightly
        rpm += (Math.random() - 0.5) * 10;
        if (rpm < 0) rpm = 0;

        const message = rpm.toFixed(2);
        client.publish(topic, message, () => {
            console.log(`Published ${message} to ${topic}`);
        });
    }, 2000);
});

client.on("error", (err) => {
    console.error(err);
});

File: sim_mqtt.py
import paho.mqtt.client as mqtt
import time
import random

broker = "broker.emqx.io"
port = 1883
topic = "mi_proyecto/sensor/rpm"

client = mqtt.Client()
client.connect(broker, port, 60)

rpm = 120.0
print("Starting MQTT simulation...")
while True:
    rpm = rpm + random.uniform(-5.0, 5.0)
    if rpm < 0: rpm = 0
    client.publish(topic, str(round(rpm, 2)))
    print(f"Published: {rpm:.2f}")
    time.sleep(2)
